### ISSUE TITLE
[BUG] Rule does not move when different folder is specified

### ISSUE THREAD

**Issue by jupiler** [Checkmk/ansible-collection-checkmk.general#683]
Verify first that your issue is not already reported [here](https://github.com/Checkmk/ansible-collection-checkmk.general/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc).
Where possible also test if the latest release and main branch are affected too.
Complete all sections as described!

**Describe the bug**
Rules are not moved to a different folder.

**Component Name**
Component Name: rule

**Ansible Version**
```console
ansible-playbook [core 2.17.4]
  config file = ~/ansible.cfg
  configured module search path = ['/home/user/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /opt/user/.venv/lib/python3.12/site-packages/ansible
  ansible collection location = /opt/user/.venv/collections
  executable location = /opt/user/.venv/bin/ansible-playbook
  python version = 3.12.3 (main, Sep 11 2024, 14:17:37) [GCC 13.2.0] (/opt/user/.venv/bin/python)
  jinja version = 3.1.4
  libyaml = True
```

**Checkmk Version and Edition**

```console
Checkmk Enterprise Edition 2.2.0p27

```

**Collection Version**

```console
$ ansible-galaxy collection list
---
collections:
  # https://github.com/Checkmk/ansible-collection-checkmk.general/tree/main
  - name: checkmk.general
    version: 5.3.0
  # https://github.com/lvrfrc87/git-acp-ansible/tree/development/ansible_collections/lvrfrc87/git_acp
  - name: lvrfrc87.git_acp
    version: 2.2.0

roles: []
```

**To Reproduce**
see playbook below

**Expected behavior**

When a folder is changed in the rule definition that the rule is moved to the correct folder

**Actual behavior**

Change of folder is not detected and does not trigger an update of the rule.

**Minimum reproduction example**

```yaml
- hosts: localhost
  gather_facts: no
  vars:
    checkmk_rule_host: "{{ checkmk_rule_host }}"
    checkmk_rule_site: "{{ checkmk_rule_site }}"
    checkmk_rule_user: "{{ checkmk_rule_user }}"
    checkmk_rule_pass: "{{ checkmk_rule_pass }}"
    checkmk_rule_ruleset: "service_label_rules"
    checkmk_rule_folder1: "/test1"
    checkmk_rule_folder2: "/test2"
    checkmk_validate_cert: false
    rule:
      host_names: ["test"]
      host_tags: []
      service_labels: []
      description: "Test rule"
      disabled: false
      checkmk_rule_ruleset: "``"
      value_raw: '{ ''testlabel'': ''value''}'
  tasks:
    - name: State_present | Create rule for {{ __present_file }} in {{ checkmk_rule_ruleset }}
      checkmk.general.rule:
        server_url: "{{ checkmk_rule_host }}"
        site: "{{ checkmk_rule_site }}"
        automation_user: "{{ checkmk_rule_user }}"
        automation_secret: "{{ checkmk_rule_pass }}"
        ruleset: "{{ checkmk_rule_ruleset }}"
        rule:
          rule_id: "{{ ruleid_info.content.id |default('') }}"
          conditions:
            host_labels: "{{ rule.host_labels | default([]) }}"
            host_name:
              match_on: "{{ rule.host_names | default(['test']) }}"
              operator: "{{ rule.host_labels | default('one_of') }}"
            host_tags: "{{ rule.host_tags | default([]) }}"
            service_labels: "{{ rule.service_labels | default([]) }}"
          properties:
            comment: Created with Ansible pipeline
            description: "{{ rule.host_names[0] }} - {{ rule.description | default('no description specified')}}"
            disabled: "{{ rule.disabled | default('False') | bool}}"
          value_raw: "{{ rule.value_raw | string }}"
          location:
            folder: "{{ item }}"
            position: "top"
        state: present
        validate_certs: "{{ checkmk_validate_cert }}"
      register: ruleid_info
      with_items:
        - "{{ checkmk_rule_folder1 }}"
        - "{{ checkmk_rule_folder2 }}"
```

**Additional context**

ok: [localhost] => (item=/test2) => {
    "ansible_loop_var": "item",
    "changed": false,
    "content": {
        "domainType": "rule",
        "extensions": {
            "conditions": {
                "host_labels": [],
                "host_name": {
                    "match_on": [
                        "test"
                    ],
                    "operator": "one_of"
                },
                "host_tags": [],
                "service_labels": []
            },
            **"folder": "/test1",**
            "folder_index": 0,
            "properties": {
                "comment": "Created with Ansible pipeline",
                "description": "test - Test rule",
                "disabled": false
            },
            "ruleset": "service_label_rules",
            "value_raw": "{'testlabel': 'value'}"
        },
        "id": "17d0d00a-d990-4cfb-848c-1df85c67a6d7",
        "links": [
            {
                "domainType": "link",
                "href": "http://checkmk/monitoring/check_mk/api/1.0/objects/rule/17d0d00a-d990-4cfb-848c-1df85c67a6d7",
                "method": "GET",
                "rel": "self",
                "type": "application/json"
            },
            {
                "domainType": "link",
                "href": "http://checkmk/monitoring/check_mk/api/1.0/objects/rule/17d0d00a-d990-4cfb-848c-1df85c67a6d7",
                "method": "DELETE",
                "rel": "urn:org.restfulobjects:rels/delete",
                "type": "application/json"
            }
        ],
        "members": {},
        "title": "test - Test rule"
    },
    "etag": "\"fb9265d344a27b574b464e0048da1c7557b914a7\"",
    "http_code": 200,
    "invocation": {
        "module_args": {
            "automation_secret": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "automation_user": "jupi",
            "rule": {
                "conditions": {
                    "host_labels": [],
                    "host_name": {
                        "match_on": [
                            "test"
                        ],
                        "operator": "one_of"
                    },
                    "host_tags": [],
                    "service_labels": []
                },
                "location": {
                    **"folder": "/test2",**
                    "neighbour": null,
                    "position": "top"
                },
                "properties": {
                    "comment": "Created with Ansible pipeline",
                    "description": "test - Test rule",
                    "disabled": false
                },
                "rule_id": "17d0d00a-d990-4cfb-848c-1df85c67a6d7",
                "value_raw": "{ 'testlabel': 'value'}"
            },
            "ruleset": "service_label_rules",
            "server_url": "https://checkmk",
            "site": "monitoring",
            "state": "present",
            "validate_certs": false
        }
    },
    "item": "/test2",
    "msg": "Rule already exists with the desired parameters."
}

    "msg": "Rule already exists with the desired parameters."
}

**Comment by msekania** [comment]
@jupiler,

what happens if you split your "loop" task in two?

Your example is doing something else and not what you intend to achieve with it.

**Comment by msekania** [comment]
There is another bug though,

It rule module does not return content of the created rule unlike what documentation states

**Comment by msekania** [comment]
@jupiler,

OK, something is weird, I can also reproduce with two tasks

**Comment by robin-checkmk** [comment]
@msekania thank you for looking into this! Does #689 fix this issue, or is it an intermediate step or even unrelated?

**Comment by msekania** [comment]
@robin-checkmk, yes!

**Comment by robin-checkmk** [comment]
Then this was fixed with `5.3.1`. :adhesive_bandage:
